src/driver/sqlite/SqliteDriver.ts

```diff
@@ -77,11 +80,55 @@ export class SqliteDriver extends AbstractSqliteDriver {
     // Protected Methods
     // -------------------------------------------------------------------------
 
+    protected createDatabaseFile(): Promise<void> {
+        return new Promise<void>((resolve, reject) => {
+            mkdirp(dirname(this.options.database), (err: NodeJS.ErrnoException) => {
+                if (err) {
+                    return reject(err);
+                }
+
+                open(this.options.database, "w", (err, fd) => {
+                    if (err) {
+                        return reject(err);
+                    }
+
+                    close(fd, (err) => {
+                        if (err) {
+                            return reject(err);
+                        }
+                    });
+
+                    return resolve();
+                });
+            });
+        });
+    }
+
+    protected doesFileExist(): Promise<boolean> {
+        return new Promise<boolean>((resolve, reject) => {
+            stat(this.options.database, (err, stats) => {
+                if (err) {
+                    if (err.code === "ENOENT") {
+                        return resolve(false);
+                    }
+
+                    return reject(err);
+                }
+
+                return resolve(stats.isFile());
+            });
+        });
+    }
+
```

Comment:
both those 2 functions looks like they can be extracted into separate class related to filesystem (not sqlite anyhow) and then used there.

Reply:
Maybe PlatformUtils is that place...